### PR TITLE
DINT-245: searchRequest and searchResponse can be joined by the same search request id

### PR DESCRIPTION
**Pull request by aniham**

### CODE REVIEW

File: packages/storefront-events-collector/src/handlers/search/searchResponseReceivedAEP.ts [hunk 1]
```diff
@@ -42,6 +42,10 @@ const handler = async (event: Event): Promise<void> => {
         value: 1,
     };
 
+    if (searchResultsCtx) {
+        payload.searchResponse.id = searchResultsCtx.data.searchRequestId as string;
```
Comment (grahamcrackers) [inline packages/storefront-events-collector/src/handlers/search/searchResponseReceivedAEP.ts:46]: why are we casting this as a string? is it not defined in the types anywhere?

File: packages/storefront-events-collector/src/handlers/search/searchRequestSentAEP.ts [hunk 2]
```diff
@@ -61,6 +62,10 @@ const handler = async (event: Event): Promise<void> => {
         value: 1,
     };
 
+    if (searchInputCtx) {
+        payload.searchRequest.id = searchInputCtx.data.searchRequestId as string;
```
Comment (grahamcrackers) [inline packages/storefront-events-collector/src/handlers/search/searchRequestSentAEP.ts:66]: why are we casting this as a string? is it not defined in the types anywhere?

File: packages/storefront-events-collector/src/handlers/search/searchRequestSentAEP.ts [hunk 3]
```diff
@@ -11,12 +11,13 @@ const handler = async (event: Event): Promise<void> => {
     const { searchUnitId, searchInputContext, debugContext, customContext } = event.eventInfo;
 
     let payload: BeaconSchema;
+
+    const searchInputCtx = createSearchInputCtx(searchUnitId as string, searchInputContext);
```
Comment (grahamcrackers) [inline packages/storefront-events-collector/src/handlers/search/searchRequestSentAEP.ts:15]: why are we casting this as a string? is it not defined in the types anywhere?

**Review by grahamcrackers — COMMENTED** [review]
left a few questions in the files

File: packages/storefront-events-collector/src/handlers/search/searchRequestSentAEP.ts (hunk 3 above)
Comment (aniham) [reply packages/storefront-events-collector/src/handlers/search/searchRequestSentAEP.ts:15]: @grahamcrackers i did not chase it down, my change was moving this existing line from outside the `else` statement

File: packages/storefront-events-collector/src/handlers/search/searchRequestSentAEP.ts (hunk 2 above)
Comment (aniham) [reply packages/storefront-events-collector/src/handlers/search/searchRequestSentAEP.ts:66]: @grahamcrackers i did not chase it down, my change was moving this existing line from elsewhere

File: packages/storefront-events-collector/src/handlers/search/searchResponseReceivedAEP.ts (hunk 1 above)
Comment (aniham) [reply packages/storefront-events-collector/src/handlers/search/searchResponseReceivedAEP.ts:46]: @grahamcrackers i did not chase it down, my change was moving this existing line from elsewhere

**Review by grahamcrackers — APPROVED** [review]
👍

**Pull request merged**